frontend/test/metabase/scenarios/visualizations/pivot_tables.cy.spec.js

```diff
@@ -0,0 +1,75 @@
+import {
+  signInAsAdmin,
+  restore,
+  openOrdersTable,
+  popover,
+} from "__support__/cypress";
+
+describe("scenarios > visualizations > pivot tables", () => {
+  beforeEach(() => {
+    restore();
+    signInAsAdmin();
+  });
+
+  it("should be created from an ad-hoc question", () => {
+    cy.server();
+    cy.route("POST", "/api/dataset").as("dataset");
+    cy.route("POST", "/api/advanced_computation/pivot/dataset").as(
+      "pivotTableDataset",
+    );
+
+    openOrdersTable({ mode: "notebook" });
+    cy.findByText("Summarize").click();
+    cy.findByText("Count of rows").click();
+    cy.findByText("Pick a column to group by").click();
+    popover().within(() => {
+      cy.findByText(/Orders?$/).click(); // Collapse "Orders" to avoid additional scrolling
+      cy.findByText(/Users?$/).click();
+    });
+    cy.get(".ReactVirtualized__Grid").scrollTo("bottom"); // "Source" is not visible - we have to scroll
+    cy.findByText("Source").click();
+    // Add another metric
+    cy.get(".Icon-add")
+      .last() // This is fragile, but there's no other way to get to this element right now
+      .click();
+    popover().within(() => {
+      cy.findByText(/Products?$/).click();
+      cy.findByText("Category").click();
+    });
+    cy.findByText("Visualize").click();
+    cy.wait("@dataset");
+
+    cy.findByText("Visualization").click();
+    cy.get(".Icon-pivot_table").click();
+    cy.wait("@pivotTableDataset");
```

Comment:
Why do we need to wait for this rather than looking for the UI?

Reply:
Just as a precaution if it takes longer for calculation to finish in CI. Relying on UI leads to all sorts of issues and race conditions.
This simple test could probably go without waiting for anything, though.

Reply:
No more waiting :)